packages/xstate-solid/src/deriveServiceState.ts

```diff
@@ -0,0 +1,44 @@
+import type { AnyState } from 'xstate';
+import type { CheckSnapshot } from './types';
+
+function isState(state: any): state is AnyState {
+  return (
+    typeof state === 'object' &&
+    'value' in state &&
+    'context' in state &&
+    'event' in state &&
+    '_event' in state &&
+    'can' in state &&
+    'matches' in state
+  );
+}
+
+/**
+ * Takes in an interpreter or actor ref and returns a State object with reactive
+ * methods or if not State, the initial value passed in
+ * @param state {AnyState | unknown}
+ * @param prevState {AnyState | unknown}
+ */
+export const deriveServiceState = <
+  StateSnapshot extends AnyState,
+  StateReturnType = CheckSnapshot<StateSnapshot>
+>(
+  state: StateSnapshot | unknown,
+  prevState?: StateSnapshot | unknown
+): StateReturnType => {
+  if (isState(state)) {
+    return {
+      ...state,
+      toJSON: state.toJSON,
+      toStrings: state.toStrings,
+      can: state.can,
+      hasTag: state.hasTag,
```

Comment:
This should be equivalent, right?
```suggestion
      ...state,
```

Reply:
These methods are on the prototype so we need to explicitly add them, they don't get add on the spread

Reply:
👍 yeah, makes sense - I forgot that they might not be enumerable